inc/stateinventory.class.php

```diff
@@ -116,16 +129,29 @@ public function display($options = [])
         echo "</tr>";
         echo "</thead>";
 
-        $sql = "SELECT `glpi_plugin_glpiinventory_taskjobstates`.*
-            FROM `glpi_plugin_glpiinventory_taskjobstates`
-         LEFT JOIN `glpi_plugin_glpiinventory_taskjobs`
-            ON `plugin_glpiinventory_taskjobs_id` = `glpi_plugin_glpiinventory_taskjobs`.`id`
-         WHERE `method` = 'networkinventory'
-         GROUP BY `uniqid`
-         ORDER BY `uniqid` DESC
-         LIMIT " . intval($start) . ", " . intval($_SESSION['glpilist_limit']);
-        $result = $DB->query($sql);
-        while ($data = $DB->fetchArray($result)) {
+        $iterator = $DB->request([
+            'SELECT' => [
+                'glpi_plugin_glpiinventory_taskjobstates.*'
+            ],
+            'FROM'   => 'glpi_plugin_glpiinventory_taskjobstates',
+            'LEFT JOIN' => [
+                'glpi_plugin_glpiinventory_taskjobs' => [
+                    'ON' => [
+                        'glpi_plugin_glpiinventory_taskjobs' => 'id',
+                        'glpi_plugin_glpiinventory_taskjobstates' => 'plugin_glpiinventory_taskjobs_id'
+                    ]
+                ]
+            ],
+            'WHERE'  => [
+                'method' => 'networkinventory'
+            ],
+            'GROUPBY' => 'uniqid',
+            'ORDERBY' => 'uniqid DESC',
+            'LIMIT'  => (int)$_SESSION['glpilist_limit'],
+            'START'  => (int)$start
+        ]);
+
```

Comment:
Some warning from this query  (GLPIInventory > Networking > Networking Inventory status)

![image](https://user-images.githubusercontent.com/7335054/236145782-6a7ae576-2389-4f89-b653-d626fef463d5.png)

```
[2023-05-04 10:04:06] glpiphplog.CRITICAL:   *** Uncaught Exception DivisionByZeroError: Division by zero in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/stateinventory.class.php at line 236
  Backtrace :
  plugins/glpiinventory/front/stateinventory.php:50  PluginGlpiinventoryStateInventory->display()
  public/index.php:82                                require()
```

Reply:
3cae3eb2dc1eccd18831a7c64dd8bca0e5ed74c6 should fix the SQL issue, dunno if that will fix the division by 0 also